mindgraph/graph.py

```diff
@@ -4,109 +4,109 @@
-        elif type(newnode) is not Node:
+    def append(self, newTask) -> "Task":
+        """ Creates a new Task and appends it to subtasks """
+        if type(newTask) is str:
+            newTask = Task(newTask)
+        elif type(newTask) is not Task:
             raise TypeError
-        self._threads.append(newnode)
-        return newnode
+        self._subtasks.append(newTask)
+        return newTask
 
-    def pop(self, index: int) -> "Node":
-        """ Pops the Node from threads[index] """
-        return self._threads.pop(index)
+    def pop(self, index: int) -> "Task":
+        """ Pops the Task from subtasks[index] """
+        return self._subtasks.pop(index)
 
-    def blockedby(self, node: "Node") -> None:
-        """ Adds a Node to the dependenceis list """
-        if type(node) is Node:
-            self._dependencies.append(node)
+    def blockedby(self, task: "Task") -> None:
+        """ Adds a task to the subtasks list """
+        if type(task) is Task:
+            self._blockers.append(task)
             return None
         else:
             raise TypeError
 
-    def blocking(self, node: "Node") -> None:
-        """ Adds this Node to another node's dependencies list """
-        if type(node) is Node:
-            node._dependencies.append(self)
+    def blocking(self, task: "Task") -> None:
+        """ Adds this task to another task's blockers list """
+        if type(task) is Task:
+            task._blockers.append(self)
             return None
         else:
             raise TypeError
 
-    def __getitem__(self, key: int) -> "Node":
-        return self._threads[key]
+    def __getitem__(self, key: int) -> "Task":
+        return self._subtasks[key]
 
     def __repr__(self) -> str:
         return '\n'.join(self.format_tree())
 
-    def format_tree(self: "Node", depth: int = 0) -> Iterator[str]:
-        """Format node and dependents in tree format, emitting lines
+    def format_tree(self: "Task", depth: int = 0) -> Iterator[str]:
+        """Format Task and dependents in tree format, emitting lines
 
-        Assumes no cycles in graph
+        Assumes no cycles in Project
         """
         indent = '    ' * depth
         bullet = '- ' if depth != 0 else ''
-        suffix = ':' if self.threads else ''
+        suffix = ':' if self.subtasks else ''
         line = '{indent}{bullet}{self.name}{suffix}'.format(**locals())
 
         yield line
-        for n in self.threads:
+        for n in self.subtasks:
             yield from n.format_tree(depth+1)
 
     def _postorder(self,
                    depth: int = 0,
-                   visited: Set["Node"] = None,
-                   node_key: Callable[["Node"], Any]=None,
-                   ) -> Generator[Tuple[int, "Node"], None, Set["Node"]]:
-        """Post-order traversal of graph rooted at node"""
+                   visited: Set["Task"] = None,
+                   Task_key: Callable[["Task"], Any]=None,
```

Comment:
another find/replace error. `Task_key` --> `task_key` (and I should probably find a clearer name (or at least document it's purpose) in the future.

Reply:
fixed in 77b8f69